### ISSUE TITLE
workerpool does not wait for child processes to terminate

### ISSUE THREAD

**Issue by boneskull** [josdejong/workerpool#175]
Hi,

It looks like when `Pool#terminate()` is called when using the `child_process` builtin, `ChildProcess#kill()` is used.  But this does not mean the child process _has actually exited_.  The current behavior may be fine in most situations--fire-and-forget--but I need to ensure the child processes have ended before moving forward.  Waiting _will_ have performance impact, so if this were to change, perhaps it's best as an option?

Anyhow, changing this relevant code in `WorkerHandler#terminate` seems to work, assuming `kill()` works as intended.  

```js
      if (typeof this.worker.kill === 'function') {
        this.worker.kill();  // child process
        this.worker.on('exit', () => {
          this.worker = null;
          this.terminating = false;
          this.terminated = true;
          if (this.terminationHandler) {
            this.terminationHandler(this);
          }
        })
        return; 
      }
```

I note that the return value of `this.worker.kill()` above is ignored; if the signal fails to send, `ChildProcess#kill` returns `false`--but there's no error-handling of that situation in `workerpool`.  Couldn't tell you what _should_ happen in that case, but just pointing it out. :smile:

This seems to differ from Node.js' `Worker#terminate()`, which looks like it's guaranteed to end the worker thread.

**Comment by josdejong** [comment]
Thanks for reporting, there are indeed options to improve the `terminate` method.

I think this is a duplicate of #32. Help improving this would be very welcome, I think we should give the `terminate` method as a whole some love.

**Comment by boneskull** [comment]
closing in lieu of #32

**Comment by josdejong** [comment]
This should be fixed now in `v6.0.2` thanks to #176
